### ISSUE TITLE
SSH Backup Stuck Syncing Cache

### ISSUE THREAD

**Issue by bettyvschmartz** [borgbackup/borg#5232]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Bug/Issue

#### Your borg version (borg -V).

1.1.7 and 1.1.5

#### Operating system (distribution) and version.

QNAP NAS and Ubuntu 18.04

#### How much data is handled by borg?

3TBs

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

borg create -s --progress user1@computer1:/media/user1/backup/borg::18-06-20 Backup_folder1/

## Describe the problem you're observing.

Using a QNAP NAS to back to an eternal HDD. The backup is ~3TB. Originally seeded the backup locally using the NAS and the drive connected over USB 3.0. Then I moved the drive to an external location with the plan to backup over SSH. Because this backup was as the root user, I couldn't access the repo in the remote location as I'm using SSH as user1. I had to take ownership of the repo as user1 then could list the backup archives and view contents.

Now when I try run a backup I'm getting:

`0% Syncing chunks cache. Processing archive 15-06-20`

Which doesn't seem to progress. I decided to issue `borg delete --cache-only` to rebuild but it's still the same; stuck on zero percent. If I issue a `du` on the cache directory I see it has created as sub directory for the cache but it's only 128KB and this seems to stay the same, i.e. not growing. 

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

Yes, by issuing the above command. 

#### Include any warning/errors/backtraces from the system logs

No errors.

**Comment by infectormp** [comment]
Can you update borg to the latest version and check again?

**Comment by bettyvschmartz** [comment]
Sure. Will try that.

Currently brought the drive locally to test running the backup and re-building the cache. This time the cache synced quickly over USB 3.0 and is 410MB but now the backup is taking an age. It's running as though it's taking a _full_ backup yet I know it isn't as `du` on the borg repo is showing no increase in size.

Output example:
`110 GB O 110 GB C 755 kB D 149 N`

This is incrementing MB-by-MB whereas usually that shoots up quickly for data that is already backed up. Is this due to deleting the local cache? Is this expected? It's as slow as running a _full_ backup.

-- BVS

**Comment by ThomasWaldmann** [comment]
@bettyvschmartz that sounds like the files cache produces many misses, then it chunks all contents and then it notices it already has most of the chunks due to deduplication.

Did you mount the source data files at a different path than usually?

Did you use a different user to do the backup? 

Either there is no files cache or pathes do not match or there is an issue with inode number stability.

**Comment by ThomasWaldmann** [comment]
Oh, you deleted the cache. That explains it, you do not have a files cache, so re-chunking is expected behaviour.

**Comment by bettyvschmartz** [comment]
Thanks for the reply. So I assume now it is scanning through the files to rebuild the cache? Is that expected to be as slow as a full backup?

I realised what I think was the issue with the backup over SSH and the lack of progress. When I closed the shell session after the first local back up I had manually set BORG_CACHE_DIR but when I ran a new shell session I never set the variable. Therefore I assume Borg was using the default location which had no cache. 

-- BVS

**Comment by ThomasWaldmann** [comment]
It is slower than usual, but faster than a full backup because it reads, chunks, hashes, looks up all content chunks, but it does not need to transfer / store them to the repo though.

**Comment by bettyvschmartz** [comment]
OK, thanks. The original backup of 3TB took around 20 hours. I've done the calculations and this looks like it will take about 15 hours which is faster than a full backup but not by much. 

Anyhow I'll leave it to complete. 

Thanks. 

-- BVS

**Comment by bettyvschmartz** [comment]
@ThomasWaldmann the original issue of the backup stuck syncing is now resolved. After diagnostic, thought I would share the fix for others who may come across the same issue. 

Borg backup was running to a remote server over VPN. When running the backup locally all was fine. I managed to identify that it was in fact the SSH session that was hanging, and not Borg. Trying SSH config options like `ClientAliveInterval` and `ClientAliveCountMax` didn't work to keep the connection alive. It transpired that the problem was due to inconsistent MTU sizes on the path between the localhost and remote server. Setting `mssfix 1400` in the openvpn config resolved this. 

-- BVS
